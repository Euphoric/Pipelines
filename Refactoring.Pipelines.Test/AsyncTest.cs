using System;
using System.Collections.Generic;
using System.Text;
using System.Threading;
using ApprovalTests;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using Refactoring.Pipelines;

namespace Refactoring.PipelinesAsync.Test
{
    [TestClass]
    public class AsyncTest
    {
        [TestMethod]
        public void ParallelRuns()
        {
            var inputPipe = new InputPipe<List<int>>("inputs");
            var listAddPipe1 = inputPipe.Process(l => AddToList(l, 1));
            var listAddPipe2 = inputPipe.Process(l => AddToList(l, 2));
            var listAddPipe3 = inputPipe.Process(l => AddToList(l, 3));
            var listAddPipe4 = inputPipe.Process(l => AddToList(l, 4));
            var listAddPipe5 = inputPipe.Process(l => AddToList(l, 5));
            var listAddPipe6 = inputPipe.Process(l => AddToList(l, 6));
            var listAddPipe7 = inputPipe.Process(l => AddToList(l, 7));


            var list = new List<int>{};
            inputPipe.Send(list);
            Approvals.VerifyAll(list, "numbers");

        }

        private static readonly Random random = new Random();
        private static int AddToList(List<int> l, int value)
        {
            Thread.Sleep(random.Next(50, 100));
            l.Add(value);
            return value;
        }
    }
}
