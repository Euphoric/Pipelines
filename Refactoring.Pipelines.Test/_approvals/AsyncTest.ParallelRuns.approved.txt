numbers[0] = 1
numbers[1] = 2
numbers[2] = 3
numbers[3] = 4
numbers[4] = 5
numbers[5] = 6
numbers[6] = 7
